src/types.rs

```diff
@@ -310,7 +314,7 @@ pub struct LightClientConfig {
 	pub ttl: u64,
 }
 
-fn delay_for(target: Duration, elapsed: Duration) -> Option<Duration> {
+pub fn delay_for(target: Duration, elapsed: Duration) -> Option<Duration> {
 	(target > elapsed).then(|| target - elapsed)
 }
```

Comment:
I think `delay_for` is a weird function, as it literally is the same as [`std::time::Duration::checked_sub`](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.checked_sub).